utils/yamlConfig.js

```diff
@@ -0,0 +1,146 @@
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+const fs = require('fs');
+const path = require('path');
+const _ = require('lodash');
+const YAWN = require('yawn-yaml/cjs');
+const mergeUtils = require('./mergeUtils');
+
+function encodeKey(key) {
+  return key.replace(/[^a-zA-Z0-9]/g, char => '_x' + char.charCodeAt(0).toString(16));
+}
+
+function flattenEnvObject(obj, prefix) {
+  const result = {};
+  const path = prefix;
+  flattenEnvObjectInternal(obj, path, result);
+  return result;
+
+  function flattenEnvObjectInternal(obj, path, result) {
+    for (const key of Object.keys(obj)) {
+      const encodedKey = encodeKey(key);
+      const newPath = path ? `${path}_${encodedKey}` : encodedKey;
+      const val = obj[key];
+      if (typeof val === 'object') {
+        flattenEnvObjectInternal(val, newPath, result);
+      } else if (Array.isArray(val)) {
+        result[newPath] = val.join(',') + ',';
+      } else {
+        result[newPath] = String(val);
+      }
+    }
+  }
+}
+
+function convertConfigToEnvObj(config) {
+  return flattenEnvObject(config, 'ZWED');
```

Comment:
We have this as a parameter in the JSON code ( https://github.com/zowe/zlux-app-server/pull/182/files#diff-b29e6cc3169d78cc28c86c0484989a44db4b3443472116c21731e99896198551R104 ) so it should be a parameter here too.